src/app/o/[orgId]/projects/[projId]/layout.tsx

```diff
@@ -0,0 +1,58 @@
+import { FC, ReactNode } from 'react';
+import { Metadata } from 'next';
+import { headers } from 'next/headers';
+
+import HomeThemeProvider from 'features/home/components/HomeThemeProvider';
+import ProjHomeLayout from 'features/campaigns/layout/ProjHomeLayout';
+import BackendApiClient from 'core/api/client/BackendApiClient';
+import { ZetkinCampaign, ZetkinOrganization } from 'utils/types/zetkin';
+
+type Props = {
+  children: ReactNode;
+  params: {
+    orgId: number;
+    projId: number;
+  };
+};
+
+export async function generateMetadata({ params }: Props): Promise<Metadata> {
+  const headersList = headers();
+  const headersEntries = headersList.entries();
+  const headersObject = Object.fromEntries(headersEntries);
+  const apiClient = new BackendApiClient(headersObject);
+
+  const proj = await apiClient.get<ZetkinCampaign>(
+    `/api/orgs/${params.orgId}/campaigns/${params.projId}`
+  );
+
+  return {
+    icons: [{ url: '/logo-zetkin.png' }],
+    title: proj.title,
+  };
+}
+
+// @ts-expect-error https://nextjs.org/docs/app/building-your-application/configuring/typescript#async-server-component-typescript-error
+const MyHomeLayout: FC<Props> = async ({ children, params }) => {
+  const headersList = headers();
+  const headersEntries = headersList.entries();
+  const headersObject = Object.fromEntries(headersEntries);
+  const apiClient = new BackendApiClient(headersObject);
+
+  const org = await apiClient.get<ZetkinOrganization>(
+    `/api/orgs/${params.orgId}`
+  );
+
+  const proj = await apiClient.get<ZetkinCampaign>(
+    `/api/orgs/${params.orgId}/campaigns/${params.projId}`
+  );
+
+  return (
+    <HomeThemeProvider>
+      <ProjHomeLayout org={org} proj={proj}>
+        {children}
+      </ProjHomeLayout>
+    </HomeThemeProvider>
+  );
```

Comment:
When I played with this and attempted to go to the project page when i was logged out, and the project was not "published", it throws an error here which breaks the page. (just for context: a project can be either a "draft" or "published", this is controlled in the organizer interface under the ... menu in the top right corner of a project, under "Edit project"). 

So the solution will be to place this in a `try` and the `catch` should return `notFound()`, you can see an example of this in `src/app/o/[orgId]/joinformverified/page.tsx`